src/gui/passkeys/PasskeyImportDialog.cpp

```diff
@@ -69,33 +85,68 @@ QSharedPointer<Database> PasskeyImportDialog::getSelectedDatabase()
     return m_selectedDatabase;
 }
 
-QUuid PasskeyImportDialog::getSelectedGroupUuid()
+QUuid PasskeyImportDialog::getSelectedEntryUuid() const
+{
+    return m_selectedEntryUuid;
+}
+
+QUuid PasskeyImportDialog::getSelectedGroupUuid() const
 {
     return m_selectedGroupUuid;
 }
 
-bool PasskeyImportDialog::useDefaultGroup()
+bool PasskeyImportDialog::useDefaultGroup() const
 {
     return m_useDefaultGroup;
 }
 
+bool PasskeyImportDialog::isImportedToEntry() const
+{
+    return m_ui->importToEntryCheckbox->isChecked();
+}
+
 QString PasskeyImportDialog::getDatabaseName(const QSharedPointer<Database>& database) const
 {
     return QFileInfo(database->filePath()).fileName();
 }
 
-void PasskeyImportDialog::addGroups(const QSharedPointer<Database>& database)
+void PasskeyImportDialog::addGroups()
 {
+    if (!m_selectedDatabase) {
+        return;
+    }
+
     m_ui->selectGroupComboBox->clear();
-    for (const auto& group : database->rootGroup()->groupsRecursive(true)) {
-        if (!group || group->isRecycled() || group == database->metadata()->recycleBin()) {
+    for (const auto& group : m_selectedDatabase->rootGroup()->groupsRecursive(true)) {
+        if (!group || group->isRecycled() || group == m_selectedDatabase->metadata()->recycleBin()) {
             continue;
         }
 
         m_ui->selectGroupComboBox->addItem(group->fullPath(), group->uuid());
     }
 }
 
+void PasskeyImportDialog::addEntries()
+{
+    if (!m_selectedDatabase || !m_selectedDatabase->rootGroup()) {
+        return;
+    }
+
+    const auto group = m_selectedDatabase->rootGroup()->findGroupByUuid(m_selectedGroupUuid);
+    if (!group) {
+        return;
+    }
+
+    m_ui->selectEntryComboBox->clear();
+    for (const auto& entry : group->entries()) {
+        if (!entry || entry->isRecycled()) {
```

Comment:
Would be better to add the recycle bin as a check above: `!group || group->isRecycled()`

Reply:
It's not possible to select recycled groups. Those are not listed at all.

Reply:
Gotcha, then technically you will never get a recycled entry.